File: python_tests/test_produkt.py
"""Python-File for testing class Produkt"""

from py_bindings import Produkt, Datum

def test_produkt():
    """Funktion zum Testen eines Produktes"""
    datum = Datum(20, 5, 2025)
    produkt1 = Produkt("Bananen", 1.992, 2.35, datum, "Lebensmittel")
    produkt2 = Produkt("Kirschen", 1.992, 2.35, datum, "Lebensmittel")

    assert produkt1.get_bezeichnung() == "Bananen"
    assert abs(1.992 - produkt1.get_menge()) < 0.0001
    assert abs(2.35 - produkt1.get_preis()) < 0.0001
    assert abs((1.992 * 2.35) - produkt1.get_gesamtpreis()) < 0.0001
    assert produkt1.get_haltbarkeit().vergleiche_datum(datum) is True
    assert produkt1.get_produkt_abteilung() == "Lebensmittel"
    # assert produkt1.get_produkt_id() == 0 Only work locally
    assert produkt1.vergleiche_produkte(produkt2) is False

File: cpp_tests/test_produkt.cpp
#include "tests/test_produkt.hpp"

void TestProdukt() {
  Datum datum{1, 1, 2025};
  Produkt produkt{"Apfel", 1.99f, 3.0f, datum, "Obst"};
  Datum event{2, 2, 2022};

  assert(produkt.GetMenge() == 1.99f);
  assert(produkt.GetPreis() == 3.0f);
  assert(produkt.GetName() == "Apfel");
  assert(produkt.GetAbteilung() == "Obst");
  assert(std::abs(produkt.GetGesamtPreis() - (1.99f * 3.0f)) < 0.001f);
  assert(produkt.GetHaltbarkeit().Print() == "01.01.2025");
  assert(event.operator==(datum) == false);

  std::cout << "Alle Tests für Produkt waren erfolgreich!" << std::endl;
}